tests/test_bst.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from structures.bst import BST, BSTNode
+import pytest
+
+
+@pytest.fixture
+def create_bst():
+    btree = BST()
+
+    n1 = BSTNode(val=1)
+    n2 = BSTNode(val=2)
+    n3 = BSTNode(val=3)
+    n4 = BSTNode(val=4)
+    n5 = BSTNode(val=5)
+    n6 = BSTNode(val=6)
+
+    btree.root = n4
+    btree._size = 6
+
+    n4.left_child = n2
+    n4.right_child = n6
+
+    n2.parent = n4
+    n6.parent = n4
```

Comment:
Wouldn't it be nifty if the parent-child relationship were entirely built simply by assigning a child?  Can you think of a way of making it so?  Then you could do this:

```python
n4.left_child = n2
assert n2.parent is n4
```
And you wouldn't have to separately assign parent.